### DISCUSSION TITLE
Custom code to external plugin

### DISCUSSION THREAD

**Discussion by prog900** [osmandapp/OsmAnd#17307] [General]
Hello. 
Is it possible to create an external plugin that would handle clicking on the map in the OsmAnd program.
And downloaded data from a remote server. And then he showed them in the window. For example, a text description, or a photo.
Is it possible to embed some code of your own into an external plugin and not just static json data?
If possible, how?

Thank you.

**Comment by mnalis** [comment]
have you looked into https://github.com/osmandapp/osmand-api-demo repository?
